src/main/java/xyz/aimcup/tournament/controller/TournamentController.java

```diff
@@ -0,0 +1,25 @@
+package xyz.aimcup.tournament.controller;
+
+import java.net.URI;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import xyz.aimcup.generated.TournamentApi;
+import xyz.aimcup.generated.model.TournamentCreateRequest;
+import xyz.aimcup.tournament.data.entity.tournament.Tournament;
+import xyz.aimcup.tournament.service.tournament.TournamentServiceFacade;
+
+@RestController
+@RequiredArgsConstructor
+public class TournamentController implements TournamentApi {
```

Comment:
Tutaj też test by się przydał. Są dwie opcje - testcontainers i realny call wtedy lub przez mock mvc , ale wtedy bez adnotacji @SpringBootTest tylko z czymś w stylu @WebMvcTest(TournamentController.class).

Reply:
Większość z tych endpointów będzie zabezpieczona. Jak zmockować zalogowanego usera z daną rolą?

Reply:
Odpalenie metody nie wymaga zalogowanego usera.
W przypadku testów na testcontainers jeżeli chcemy taki test end-to-end w sensie przetestować całe działanie wraz z serwisem to wtedy po prostu w teście trzeba dodać usera na początku do kontekstu przed odpaleniem metody kontrolera.

Reply:
Wgl to można używając testcontainers chyba też używać mockito, ale to musisz doczytac. Wtedy możesz zamockować metode .createTournament